src/policyagent/policy-agent.ts

```diff
@@ -433,10 +433,20 @@ export class PolicyAgent extends EventEmitter {
   getConditionalUrl(conditionalUrlKey: string): string {
     let loginUrl = null;
     conditionalUrlKey = conditionalUrlKey.replace(/ /g, '');
+    // store queryparams first if login url has any
+    const extratQueryParams = conditionalUrlKey.split('?')[1];
+    // Keep without query parameters part
+    conditionalUrlKey = conditionalUrlKey.split('?')[0];
+    // Get conditional url values with conditions
     const conditionalUrlVal = conditionalUrlKey.split("=")[1];
+    // Split condition with actual login url
     if (conditionalUrlVal && conditionalUrlVal.indexOf("|") > -1) {
       if (this.options.appUrl.indexOf(conditionalUrlVal.split("|")[0]) > -1) {
         loginUrl = conditionalUrlVal.split("|")[1];
```

Comment:
@crazyrohila , com.forgerock.agents.conditional.login.url is multi valued property, there can be com.forgerock.agents.conditional.login.url[1], com.forgerock.agents.conditional.login.url[2] etc, correct way should be we need to get the map of domain & login url and use the correct one according to the application domain. 

`com.forgerock.agents.conditional.login.url[0]=app.example.com:8000|https://ds-acc-auth.sphdigital.com/SPHAuth/login?svc=st_online`

Reply:
Aha, my bad. I missed this one, pushed the changes to get a map of `condition => URL` and then checked for the current app in passed in config.